src/Microsoft.SqlTools.ServiceLayer/AutoParameterizaition/Exceptions/ParameterizationFormatException.cs

```diff
@@ -0,0 +1,41 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using Microsoft.SqlTools.ServiceLayer.AutoParameterizaition.Helpers;
+using System;
+
+namespace Microsoft.SqlTools.ServiceLayer.AutoParameterizaition.Exceptions
+{
+    /// <summary>
+    /// ParameterizationFormatException is used to surface format exceptions encountered in the TSQL batch to perform auto-parameterization of literals
+    /// </summary>
+    class ParameterizationFormatException : FormatException
+    {
+        public readonly int LineNumber;
+        public readonly string LiteralValue;
+        public readonly string CodeSenseMessage;
+        public readonly string LogMessage;
+        public readonly string TelemetryMessage;
+        public readonly string SqlDatatype;
+        public readonly string CSharpDataType;
+
+        public ParameterizationFormatException(MessageHelper.MessageType type, string variableName, string sqlDataType, string cSharpDataType, string literalValue, int lineNumber)
+            : this(type, variableName, sqlDataType, cSharpDataType, literalValue, lineNumber, null)
```

Comment:
named parameter for the magic constant

Reply:
renamed variable from e to exception and added the parameter hint to the function call